jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ServerShutdown.java

```diff
@@ -0,0 +1,518 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.TypeUtil;
+import org.eclipse.jetty.util.annotation.Name;
+import org.eclipse.jetty.util.component.Destroyable;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.eclipse.jetty.util.thread.AutoLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default implementation of {@link Server.Shutdown}, which will listen
+ * to a configured Host / Port and handle commands to control the
+ * list of Jetty Component {@link LifeCycle} objects.
+ *
+ * <p>
+ *   The commands you can send to this Server are always in {@code US_ASCII}
+ *   with a CRLF ({@code \r\n}.
+ * </p>
+ *
+ * <p>
+ *   Supported commands:
+ * </p>
+ *
+ * <dl>
+ * <dt>{@code FORCESTOP}</dt>
+ * <dd>Will stop components by calling {@link LifeCycle#stop()}.<br>
+ *     Will receive {@code Stopped\r\n} after stop, and before exit.
+ * </dd>
+ *
+ * <dt>{@code STOPEXIT}</dt>
+ * <dd>Will stop components by calling {@link LifeCycle#stop()},<br>
+ *     (Each component that is a {@link Destroyable} will also be {@link Destroyable#destroy() destroyed})
+ *     followed by {@code System.exit(0);} at the end.<br>
+ *     Will receive {@code Stopped\r\n} after stop/destroy, and before exit.</dd>
+ *
+ * <dt>{@code EXIT}</dt>
+ * <dd>Will simply call {@code System.exit(0);} on JVM</dd>
+ *
+ * <dt>{@code STATUS}</dt>
+ * <dd>Will return {@code OK\r\n} to indicate ServerShutdown is alive and ready to take commands.</dd>
+ *
+ * <dt>{@code PID}</dt>
+ * <dd>Will return PID of the running JVM from {@link ProcessHandle#pid() ProcessHandle.current().pid()} on JVM</dd>
+ * </dl>
+ *
+ * @since 12.0.23
+ */
+public class ServerShutdown implements Server.Shutdown
```

Comment:
Need a new class name, as it needs to imply that it is actively  monitoring for stop commands, then performing the shutdown. `ServerShutdownService` ? `ShutdownSupervisor`? `ShutdownManager`?